main.tf

```diff
@@ -15,16 +15,21 @@
  */
 
 terraform {
-  required_version = ">= 0.12"
+  # 0.12.6 introduced for_each for resources
+  required_version = ">= 0.12.6"
 }
 
 locals {
   prefix = var.prefix == "" ? "" : "${var.prefix}-"
+  names_map = zipmap(var.names, var.names)
+  folders = values(google_folder.folders)
```

Comment:
This should respect the order of the `var.names` variable, not be in lexical order.

Reply:
fixed in 5fe55f9